### PR TITLE
adding basic stats for authorization decisions

### PR DESCRIPTION
**Pull request by tulinkry**
this will provide some info measurements for the authorization performance

### CODE REVIEW

File: src/org/opensolaris/opengrok/authorization/AuthorizationFramework.java [hunk 1]
```diff
@@ -599,18 +600,28 @@ private boolean checkAll(HttpServletRequest request, String cache, Nameable enti
...
             // cache hit
+            stats.addRequest(request, "authorization_cache_hits");
             return val;
         }
 
+        stats.addRequest(request, "authorization_cache_misses");
+
+        long time = System.currentTimeMillis();
+
         boolean overallDecision = performCheck(entity, predicate);
```
Comment (vladak) [inline src/org/opensolaris/opengrok/authorization/AuthorizationFramework.java]: I'd add this to the stats too. Might be good to detect brute-forcing attacks.

Comment (tulinkry) [inline src/org/opensolaris/opengrok/authorization/AuthorizationFramework.java]: added

File: src/org/opensolaris/opengrok/authorization/AuthorizationFramework.java [hunk 2]
```diff
@@ -599,18 +600,28 @@ private boolean checkAll(HttpServletRequest request, String cache, Nameable enti
...
 
+        stats.addRequestTime(request, "authorization", System.currentTimeMillis() - time);
+        stats.addRequestTime(request, "authorization_of_" + entity.getName(), System.currentTimeMillis() - time);
```
Comment (vladak) [inline src/org/opensolaris/opengrok/authorization/AuthorizationFramework.java]: these 2 lines should probably use the same time value

Comment (tulinkry) [inline src/org/opensolaris/opengrok/authorization/AuthorizationFramework.java]: changed that already

**Pull request merged**